lib/modules.nix

```diff
@@ -58,6 +58,18 @@ rec {
             default = check;
             description = "Whether to check whether all option definitions have matching declarations.";
           };
+
+          _module.unstructuredType = mkOption {
+            # Disallow merging for now, but could be implemented nicely with a `types.optionType`
+            type = types.nullOr (types.uniq types.attrs);
```

Comment:
Is this an approximation of the type of types, with type merging disabled? I think we need `types.type` :)

Reply:
Since this here would be the only use of that type, and possibly nobody needs merging right now, I think it would be fine to leave it at this approximation for now.